dpcpp/factorization/factorization_helpers.dp.hpp

```diff
@@ -0,0 +1,110 @@
+// SPDX-FileCopyrightText: 2017 - 2024 The Ginkgo authors
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <CL/sycl.hpp>
+
+#include "core/factorization/factorization_helpers.hpp"
+#include "dpcpp/base/config.hpp"
+#include "dpcpp/base/dim3.dp.hpp"
+#include "dpcpp/base/dpct.hpp"
+#include "dpcpp/components/thread_ids.dp.hpp"
+
+
+namespace gko {
+namespace kernels {
+namespace dpcpp {
+namespace factorization {
+namespace helpers {
+
+using namespace ::gko::factorization;
```

Comment:
To make this more reasonable, I also removed the `factorization` namespace in the core file. Otherwise I would have had to use `::gko::factorization::` everywhere instead of just `helpers::`.

Reply:
sorry, I thought they are in the same namespace, but one is under kernel namespace and the other is under gko directly.
then the original one is better as it does not give any ambiguous function.
although it can only declare the two class you need, it's a bit annoying with the template.